README.md

```diff
@@ -1,161 +1,720 @@
 
-## 1.2 Insights
+## 1.3 Environments
 
-TODO: discuss single system image / subjectivity
+PomoDB is designed to be:
 
-TODO: discuss CRDTs / BFT-CRDTs
+- Run anywhere
+- Transport agnostic
+- Operate when disconnected from the network
 
-TODO: discuss location independence
+Of particular interest are operating in browsers, mobile applications, and IoT devices.
 
-TODO: discuss CALM Theorem and its connection with Datalog
+## 1.4 Classification
+
+PomoDB can be seen as a maximally-sharded schemaless database, a meta-circular CRDT framework, or both simultaneously. It decouples the raw data from interpretation, and expresses [CRDTs as database queries][Keep Calm and CRDT On]. Other than being able to benefit from the last 50+ years of database research, this flexibility makes the database itself user extensible.
 
 # 2. Design
 
-## 2.1 Types
+## 2.1 Relation
 
-PomoDB is designed to be run within WebAssembly, and so its types and their encodings are informed by the format. See the [WebAssembly specification](https://webassembly.github.io/spec/core/appendix/index-types.html) for more information.
+A relation is a set of tuples, where each component of the tuple is called an attribute, and can be referenced by an attribute name.
 
-Note, however, that only some types are supported as PomoDB primitives. These are:
+An n-ary relation contains n-tuples, which can each be written: $\langle a_1: v_1, a_2: v_2, \dots , a_n: v_n \rangle$, where $a_1, a_2, \dots , a_n$ give the names of each attribute, and $v_1, v_2, \dots , v_n$ are their values.
 
-- [Numbers](https://webassembly.github.io/spec/core/syntax/types.html#syntax-numtype)
-- [Opaque Reference Types](https://webassembly.github.io/spec/core/syntax/types.html#reference-types)
+## 2.2 Time
 
-As WebAssembly does not define common types like booleans or strings, these are handled using opaque reference types, and more information is available in the [serialization](pomo_db/serialization.md) specification.
+> Cause and effect are two sides of one fact.
+> 
+> Ralph Waldo Emerson 
 
-TODO: Update the above reference once that data exists
+PomoDB is a temporal database which internally models the progression of time as data changes, with each batch of data relating to a timestamp, called an epoch.
 
-## 2.2 Relation
+While epochs MUST be represented as an unsigned integer, runtimes MAY refine this representation in order to capture additional metadata about the progression of time.
```

Comment:
Do epochs necessarily need to be represented using an unsigned integer? Really they just need to form a partial order, and there's nothing stopping an implementation from, for example, denoting epochs using the CID of the block that describes the contents and predecessor of that epoch.

Such a representation has the benefit of allowing for branching timelines, at the epoch level, if for example you wanted to run an experiment that pulls tuples from a source that doesn't appear on the main branch.

Reply:
Yeah I think this is left over from when I still wanted more about DBSP in here. I think we should actually remove all references to epochs, and leave that to the Pomo Flow spec.